sky/data/storage.py

```diff
@@ -2858,6 +3028,10 @@ def delete(self) -> None:
         logger.info(f'{colorama.Fore.GREEN}{msg_str}'
                     f'{colorama.Style.RESET_ALL}')
 
+    def remove_objects_from_sub_path(self) -> None:
+        assert self._bucket_sub_path is not None, 'bucket_sub_path is not set'
+        raise NotImplementedError('Not implemented')
```

Comment:
I didn't find the documentation for the SDK we use for COS, so I couldn't find the corresponding way to delete a subdirectory. 
I might also need your help @romilbhardwaj @Michaelvll